core/src/main/java/io/micronaut/core/util/RegexPathMatcher.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.util;
+
+import java.util.regex.Pattern;
+
+/**
+ * PathMatcher implementation for regex-style patterns.
+ * @author Rafael Acevedo
+ * @since 3.0
+ */
+public class RegexPathMatcher implements PathMatcher {
+
+  @Override
+  public boolean matches(String pattern, String source) {
+    Pattern compiledPattern = Pattern.compile(pattern);
```

Comment:
Is it necessary to keep a cache of pre-compiled patterns here? @jameskleeh

Reply:
I think so yes

Reply:
done!